x-pack/plugins/spaces/server/routes/api/public/index.ts

```diff
@@ -4,17 +4,48 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { Logger, Headers } from 'src/core/server';
+import { SavedObjectsService } from 'src/legacy/server/kbn_server';
+import { XPackMainPlugin } from '../../../../../xpack_main/xpack_main';
 import { routePreCheckLicense } from '../../../lib/route_pre_check_license';
 import { initDeleteSpacesApi } from './delete';
 import { initGetSpacesApi } from './get';
 import { initPostSpacesApi } from './post';
 import { initPutSpacesApi } from './put';
+import { SpacesServiceSetup } from '../../../new_platform/spaces_service/spaces_service';
+import { SpacesHttpServiceSetup } from '../../../new_platform/plugin';
 
-export function initPublicSpacesApi(server: any) {
-  const routePreCheckLicenseFn = routePreCheckLicense(server);
+type InterfaceExcept<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
```

Comment:
It's wide spread pattern to call it `Omit` type.
even in our codebase. https://github.com/restrry/kibana/blob/a0bd3787566b0ade33b64f30927611d459a4f214/x-pack/plugins/infra/public/lib/lib.ts#L180 
Let's rename it or even declare it on the project level.

Reply:
Thanks for letting me know! I [stole this from Beats CM](https://github.com/elastic/kibana/blob/8b892487644d8504b5f4bbe0c8d6ae3148153b4c/x-pack/plugins/beats_management/server/utils/helper_types.ts#L7), but didn't look elsewhere. I'll update to be consistent